server/src/main/java/org/apache/druid/segment/metadata/FingerprintGenerator.java

```diff
@@ -53,12 +59,19 @@ public FingerprintGenerator(ObjectMapper objectMapper)
    * Generates fingerprint or hash string for an object using SHA-256 hash algorithm.
    */
   @SuppressWarnings("UnstableApiUsage")
-  public String generateFingerprint(SchemaPayload schemaPayload, String dataSource, int version)
+  public String generateFingerprint(final SchemaPayload schemaPayload, final String dataSource, final int version)
   {
+    // Sort the column names in lexicographic order
+    // This ensures that all permutations of a given columns would result in the same fingerprint
+    // thus avoiding schema explosion in the metadata database
+    // Note that this signature is not persisted anywhere, it is only used for fingerprint computation
+    final RowSignature sortedSignature = getLexicographicallySortedSignature(schemaPayload.getRowSignature());
+    final SchemaPayload updatedPayload = new SchemaPayload(sortedSignature, schemaPayload.getAggregatorFactories());
```

Comment:
Please mention a note about the aggregator factories as well that they are column order independent since they are backed by a map.